helper_test.go

```diff
@@ -2025,6 +2027,60 @@ func createProject(t *testing.T, client *Client, org *Organization) (*Project, f
 	}
 }
 
+func createTarGzipArchive(t *testing.T, files []string, outputPath string) {
+	if len(files) == 0 {
+		t.Fatal("files to archive are empty")
+	}
+
+	out, err := os.Create(outputPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer out.Close()
+
+	gw := gzip.NewWriter(out)
+	defer gw.Close()
+
+	tw := tar.NewWriter(gw)
+	defer tw.Close()
+
+	for _, filename := range files {
+		file, err := os.Open(filename)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer file.Close()
```

Comment:
Hmm, the linter didn't like this `defer`, and I was curious why. I guess it's because it doesn't close the files until the whole function returns, even though you're done with each file at the end of its loop iteration. 

The best suggestion I found was to wrap the body of the loop in an anonymous, immediately executed function (`func() { ... }()`), so each defer will get resolved at the end of its iteration.